File: src/main/java/compiler/parser/operators/SubtractParser.java
package compiler.parser.operators;

import compiler.block.Block;
import compiler.block.operators.SubtractBlock;
import compiler.parser.Parser;
import compiler.tokenizer.Tokenizer;

public class SubtractParser extends Parser<SubtractBlock> {
    @Override
    public boolean shouldParse(String line) {
        return line.matches("[a-zA-Z][a-zA-Z0-9]* [-][=] [0-9]+");
    }

    @Override
    public SubtractBlock parse(Block superBlock, Tokenizer tokenizer) {

        String name = tokenizer.nextToken().getToken();

        tokenizer.nextToken().getToken();

        String value = tokenizer.nextToken().getToken();

        return new SubtractBlock(superBlock,name,value);
    }
}

File: src/main/scala/compiler/parser/operators/SubtractParser.scala
/*
 * Cobalt Programming Language Compiler
 * Copyright (C) 2017  Michael Haywood
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package compiler.parser.operators

import compiler.block.Block
import compiler.block.operators.SubtractBlock
import compiler.parser.Parser
import compiler.tokenizer.Tokenizer

class SubtractParser extends Parser[SubtractBlock] {
  def shouldParse(line: String): Boolean = line.matches("[a-zA-Z][a-zA-Z0-9]*[ ]*[-][=][ ]*[0-9]+")

  def parse(superBlock: Block, tokenizer: Tokenizer): SubtractBlock = {
    val name: String = tokenizer.nextToken.token
    tokenizer.nextToken
    val value: String = tokenizer.nextToken.token
    new SubtractBlock(superBlock, name, value)
  }
}
